ingenialink/ethercat/servo.py

```diff
@@ -94,13 +94,26 @@ def __init__(
     ):
         if not pysoem:
             raise pysoem_import_error
-        self.__slave = slave
+        self.__slave: CdefSlave = slave
         self.slave_id = slave_id
         self._connection_timeout = connection_timeout
         self.__emcy_observers: list[Callable[[EmergencyMessage], None]] = []
         self.__slave.add_emergency_callback(self._on_emcy)
         super().__init__(slave_id, dictionary_path, servo_status_listener)
 
+    def check_servo_is_in_preoperational_state(self) -> None:
+        """Checks if the servo is in preoperational state.
+
+        Raises:
+            ILPDOOperationalError: if servo is not in preoperational state.
```

Comment:
Now the exception name makes no sense.
ILPDONoPreOpError seems a tongue twister. A generic ILEcatStateError could work...